### PR TITLE
Upgrade to Hazelcast-Kubernetes 3.8 and updated docs

### PR DESCRIPTION
**Pull request by oscerd**
**What this PR does / why we need it**:
Upgrade to latest hazelcast-kubernetes version the example and update the docs reflecting the latest version

### CODE REVIEW

**Comment by oscerd** [comment]
@pires

**Comment by pires** [comment]
@oscerd this lgtm but I'm wondering if we should wait for @prestonvanloon issue https://github.com/pires/hazelcast-kubernetes-bootstrapper/issues/16 to be fixed. Thoughts?

**Comment by pires** [comment]
Can someone approve this and assign it to me?

**Comment by oscerd** [comment]
Yeah, no problem. I'll upgrade the PR with the next version you'll release. Thanks

**Comment by oscerd** [comment]
@pires updated.

File: examples/storage/hazelcast/README.md [hunk 1]
```diff
@@ -117,26 +117,28 @@ $ kubectl get endpoints hazelcast -o yaml
...
-  resourceVersion: "11360"
+  resourceVersion: "65060"
   selfLink: /api/v1/namespaces/default/endpoints/hazelcast
-  uid: 46447198-70eb-11e6-940c-0800278ab84d
+  uid: 62645b71-0963-11e7-b39c-080027985ce6
 subsets:
 - addresses:
-  - ip: 10.244.37.2
+  - ip: 172.17.0.2
+    nodeName: minikube
```
Comment (pires) [inline examples/storage/hazelcast/README.md]: Are the official examples tied to minikube now?

Comment (oscerd) [inline examples/storage/hazelcast/README.md]: I don't have a Kubernetes cluster to run example. I use Minikube.

Comment (pires) [inline examples/storage/hazelcast/README.md]: ah, nevermind, it's just read-only info from the api.

**Review by pires — APPROVED** [review]
/lgtm

**Comment by pires** [comment]
@k8s-bot non-cri e2e test this

**Comment by oscerd** [comment]
Maybe we need to remove do-not-merge

**Comment by pires** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by oscerd** [comment]
do-not-merge is blocking this.

**Pull request merged**